methods/array_ops/src/pg_gp/array_ops.c

```diff
@@ -1929,3 +1971,126 @@ General_Array_to_Array(
     return pgarray;
 }
 
+/******* internal functions ********/
```

Comment:
This function would be much easier to code and maintain if we used the abstraction layer + eigen to compute the matrix dot product. See closestColumnsAndDistances in metric.cpp for an example of how a matrix and a vector are used in an operation.